libs/sdk-ui-kit/src/EditableLabel/tests/EditableLabel.test.tsx

```diff
@@ -0,0 +1,152 @@
+// (C) 2007-2020 GoodData Corporation
+import React from "react";
+import { mount, ReactWrapper } from "enzyme";
+
+import { EditableLabel } from "../EditableLabel";
+import { IEditableLabelProps, IEditableLabelState } from "../typings";
+import Mock = jest.Mock;
+
+function renderEditableLabel(
+    options: IEditableLabelProps,
+): ReactWrapper<IEditableLabelProps, IEditableLabelState, EditableLabel> {
+    return mount(<EditableLabel {...options} />);
+}
+
+describe("EditableLabel", () => {
+    describe("lifecycle", () => {
+        it("should render in overlay", () => {
+            const wrapper = renderEditableLabel({
+                value: "",
+                textareaInOverlay: true,
+                onSubmit: jest.fn(),
+            });
+
+            expect(wrapper.find(".gd-editable-label-textarea-wrapper").length).toEqual(0);
+
+            wrapper.find(".gd-editable-label").simulate("click");
+
+            expect(wrapper.find(".gd-editable-label-textarea-wrapper").length).toEqual(1);
+        });
+
+        it("should have editing class", () => {
+            const wrapper = renderEditableLabel({
+                value: "",
+                onSubmit: jest.fn(),
+            });
+
+            wrapper.find(".gd-editable-label").simulate("click");
+            // second trigger should cancel nor change the edit mode
+            wrapper.find(".gd-editable-label").simulate("click");
+
+            expect(wrapper.find(".gd-editable-label.is-editing").length).toEqual(1);
+
+            wrapper.unmount();
+        });
+
+        it("should be one row high", () => {
+            const wrapper = renderEditableLabel({
+                value: "",
+                onSubmit: jest.fn(),
+            });
+
+            wrapper.find(".gd-editable-label").simulate("click");
+
+            expect(wrapper.find("textarea").prop("rows")).toEqual(1);
+        });
+
+        it("should not let user enter characters above the limit", () => {
+            const wrapper = renderEditableLabel({
+                value: "",
+                maxLength: 10,
+                onSubmit: jest.fn(),
+            });
+
+            wrapper.find(".gd-editable-label").simulate("click");
+
+            expect(wrapper.find("textarea").prop("maxLength")).toEqual(10);
+        });
+
+        it("should change value when receieved", () => {
+            const wrapper = renderEditableLabel({
+                value: "aaa",
+                onSubmit: jest.fn(),
+            });
+
+            wrapper.find(".gd-editable-label").simulate("click");
+            wrapper.setProps({ value: "bbb", onSubmit: jest.fn() });
+
+            expect(wrapper.find("textarea").prop("defaultValue")).toEqual("bbb");
+        });
+    });
+
+    describe("saving and canceling and change value", () => {
+        let onCancelSpy: Mock;
+        let onSubmitSpy: Mock;
+        let onChangeSpy: Mock;
+        let wrapper: ReactWrapper<IEditableLabelProps, IEditableLabelState, EditableLabel>;
```

Comment:
Can we avoid this "global" shared state between tests?

Reply:
Fixed.